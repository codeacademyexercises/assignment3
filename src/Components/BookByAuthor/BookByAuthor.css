.BookByAuthor {
    width: 70%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
}

.Books {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.BookByAuthor h1{
    padding: auto 10px auto 10px;
    background-color: white;
    color: #EE975D;
}

.Books :nth-child(n+1){
    background-color: #ED965D;
}