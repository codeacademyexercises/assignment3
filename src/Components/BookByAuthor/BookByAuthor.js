import React, { Component } from 'react';
import Book from '../Book/Book';
import './BookByAuthor.css';

class BookByAuthor extends Component {
    render(){
        let display=[];
        this.props.InitialState.forEach((book, index)=>{
            display.push(<Book className="Book" bookinfo={book}/>);
        });
        return(
        <div >
            <div className="BookByAuthor">
                <h1>{this.props.author}</h1>
                <div className="Books">
                    {display}
                </div>
            </div>
        </div>
        );
    };
};

export default BookByAuthor;