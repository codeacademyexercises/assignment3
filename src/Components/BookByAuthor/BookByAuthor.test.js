import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import BookByAuthor from './BookByAuthor';

describe('<BookByAuthor /> ',()=>{
    it('snapShot test ',()=>{
        const wrapper = shallow(<BookByAuthor InitialState={[{}]}/>);
        expect(wrapper).toMatchSnapshot();
    })
})