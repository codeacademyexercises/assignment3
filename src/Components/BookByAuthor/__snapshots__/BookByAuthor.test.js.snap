// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookByAuthor />  snapShot test  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookByAuthor
    InitialState={
      Array [
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="BookByAuthor"
      >
        <h1 />
        <div
          className="Books"
        >
          <Book
            bookinfo={Object {}}
            className="Book"
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <div
            className="Books"
          >
            <Book
              bookinfo={Object {}}
              className="Book"
            />
          </div>,
        ],
        "className": "BookByAuthor",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Book
                bookinfo={Object {}}
                className="Book"
              />,
            ],
            "className": "Books",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bookinfo": Object {},
                "className": "Book",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="BookByAuthor"
        >
          <h1 />
          <div
            className="Books"
          >
            <Book
              bookinfo={Object {}}
              className="Book"
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <div
              className="Books"
            >
              <Book
                bookinfo={Object {}}
                className="Book"
              />
            </div>,
          ],
          "className": "BookByAuthor",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Book
                  bookinfo={Object {}}
                  className="Book"
                />,
              ],
              "className": "Books",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bookinfo": Object {},
                  "className": "Book",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
